### PR TITLE
Feature/view presentations

### PR DESCRIPTION
**Pull request by natyork**
This PR creates a filter for the presentations index view, IOT filter page by presenters , per pivotal story: https://www.pivotaltracker.com/story/show/140356273

### CODE REVIEW

**Comment by jacquesporveau** [comment]
jenkins test this please

File: app/controllers/presentations_controller.rb
Comment (jacquesporveau) [inline app/controllers/presentations_controller.rb]: Brendan has asked me to move to the safe navigation operator. 

http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/

File: app/views/presentations/_filter.html.erb
Comment (jacquesporveau) [inline app/views/presentations/_filter.html.erb]: Indent.

File: app/views/presentations/index.html.erb
Comment (jacquesporveau) [inline app/views/presentations/index.html.erb]: What about the idea of having this filter at the top somewhere. The table will differ in row count making the filter show up at different places down the page. If it is above all the rows it will always been in the same place.

**Review by jacquesporveau — CHANGES_REQUESTED** [review]

File: app/views/presentations/_filter.html.erb
Comment (natyork) [reply app/views/presentations/_filter.html.erb]: I think i need to look at my vim settings, because my indents often un-indent automatically if I make changes to the line for erb files....it must be interpreting the tags as being at the same level? I'll investigate, because you've caught this a couple times.

File: app/views/presentations/index.html.erb
Comment (natyork) [reply app/views/presentations/index.html.erb]: good point, i like that idea

File: app/views/presentations/_filter.html.erb
Comment (graygilmore) [inline app/views/presentations/_filter.html.erb]: Does this allow us to also set a "All Presenters" option?

**Review by graygilmore — CHANGES_REQUESTED** [review]
The commit that adds the form cannot function without the controller changes. The commits should either be re-ordered or squashed into a single commit.

File: app/views/presentations/_filter.html.erb
Comment (natyork) [reply app/views/presentations/_filter.html.erb]: I've added in the following option to collection_select:
 `include_blank: 'All Presenters' `
This allows us to filter for All presenters in the controller.

**Comment by natyork** [comment]
Jenkins failing due to Rubocop Ruby parser defaulting to v 2.1. PR #55 updates the .rubocop.yml file to set TargetRubyVersion to 2.3.1. When #55 is merged, there should be no rubocop offenses for the safe navigation operator.

**Comment by qr8r** [comment]
@natyork Since PR #55 is blocking future PRS, can someone extract the changes required to rubycop into their own pr that I can merge separately from the new features being added there?

File: app/controllers/presentations_controller.rb
Comment (qr8r) [reply app/controllers/presentations_controller.rb]: I would also take a look at the methods ruby hash probides you for doing this.

#fetch and #dig

**Comment by qr8r** [comment]
This is great! It definitely works. I mentioned the other day there is a fairly common rails tool for implementing this. I was wondering if you had been able to find it.

**Comment by natyork** [comment]
@brendandeere I did look at Ransack as an option, but I thought it might be more than we needed for this app. However, if you think that is a more sensible/flexible approach, I could use that instead!

**Comment by qr8r** [comment]
Yeah, I think I would like to see Ransack implemented. Or at the very least if its not something we wanted to do now,  I would like to see the ActiveRecord methods moved out of the controller and into a scope on Presentation.

**Comment by natyork** [comment]
@brendandeere, this PR has been updated to make use of Ransack for the filter feature.

File: app/controllers/presentations_controller.rb
Comment (jacquesporveau) [inline app/controllers/presentations_controller.rb]: I'd like this instance variable named a little bit more intuitively. I am guessing it means query but I don't know if that's what it is.

**Review by jacquesporveau — DISMISSED** [review]
Would also like to see the commits broken up a little. Maybe there is an instance in the future where we want to keep the ransack gem installed but ditch the implementation. Things like that.

File: app/controllers/presentations_controller.rb
Comment (natyork) [reply app/controllers/presentations_controller.rb]: It's a ransack default search parameter. You can customize it, but I left the default for the time being to minimize query string length.
https://github.com/activerecord-hackery/ransack/tree/v1.8.2#simple-mode

**Review by graygilmore — APPROVED** [review]
Looks good on my end. Would love to know what @brendandeere thinks as my ransacking is not up to date.

**Comment by natyork** [comment]
Split into two commits and updated the search key to something more readable per @jacquesporveau suggestions . Waiting on @brendandeere review before merging.

**Review by qr8r — APPROVED** [review]

**Pull request merged**
